lib/mpool/src/mpool.c

```diff
@@ -405,63 +407,36 @@ mpool_props_get(struct mpool *mp, struct mpool_props *props)
     memset(props, 0, sizeof(*props));
 
     for (i = MP_MED_BASE; i < MP_MED_COUNT; i++) {
-        struct mpool_mclass_props mcp = {0};
         merr_t                    err;
 
-        err = mpool_mclass_props_get(mp, i, &mcp);
+        err = mpool_mclass_props_get(mp, i, &props->mclass[i]);
         if (err) {
             if (merr_errno(err) == ENOENT)
                 continue;
             return err;
         }
-
-        props->mp_mblocksz[i] = mcp.mc_mblocksz;
     }
 
     return 0;
 }
 
 merr_t
-mpool_stats_get(struct mpool *mp, struct mpool_stats *stats)
+mpool_info_get(struct mpool *mp, struct mpool_info *info)
 {
-    uint64_t fsid[MP_MED_COUNT] = {};
-
-    if (!mp || !stats)
+    if (!mp || !info)
         return merr(EINVAL);
 
-    memset(stats, 0, sizeof(*stats));
+    memset(info, 0, sizeof(*info));
 
     for (int i = MP_MED_BASE; i < MP_MED_COUNT; i++) {
-        struct mpool_mclass_stats mcs = {};
-        merr_t                    err;
-        bool                      uniqfs = true;
+        merr_t err;
 
-        err = mpool_mclass_stats_get(mp, i, &mcs);
+        err = mpool_mclass_info_get(mp, i, &info->mclass[i]);
         if (err) {
             if (merr_errno(err) == ENOENT)
                 continue;
             return err;
         }
-
-        stats->mps_allocated += mcs.mcs_allocated;
-        stats->mps_used += mcs.mcs_used;
-        stats->mps_mblock_cnt += mcs.mcs_mblock_cnt;
-
-        strlcpy(stats->mps_path[i], mcs.mcs_path, sizeof(stats->mps_path[i]));
-
-        fsid[i] = mcs.mcs_fsid;
-
-        for (int j = i; j >= MP_MED_BASE; j--) {
```

Comment:
I presume that the uniqfs check is now moved to the CLI layer when reporting stats for "hse storage info"?